src/ciphers/sha256.rs

```diff
@@ -1,21 +1,61 @@
-//! SHA-2 (256 Bit)
-
+/// Represents the state of the buffer used in the SHA256 algorithm.
 struct BufState {
+    // The data to process.
     data: Vec<u8>,
+    // The length of the data.
     len: usize,
+    // The total length of the data.
     total_len: usize,
+    // A flag indicating if the buffer has a single byte.
     single: bool,
+    // A flag indicating if the buffer has been fully processed.
```

Comment:
It's fine here since this struct is private, but note that these should be doc comments with `///`, i.e., fields should also be documented in docs.